### PR TITLE
feat(VisualStudio2022): Update the Plugin to VS2022 Version for x64

### PR DESCRIPTION
**Pull request by felix1000**
It is a very quick update to let this extension can run on VS2022.
Take if it is useful

### CODE REVIEW

**Comment by unchase** [comment]
@felix1000 what tooling did you use?

**Comment by henning-krause** [comment]
@felix1000 I just tried to get your PR to work on my installation. I had to change this line to get the VSIX installer to install the new extension:

```xml
<Prerequisite Id="Microsoft.VisualStudio.Component.CoreEditor" Version="[15.0,17.0)" DisplayName="Visual Studio core editor" />
```

The installation succeeded, but I don't see the Odata option in the Connected Services view. However, Visual studio lists the extension as installed.

Any ideas?

**Comment by Odraio** [comment]
Any news about this pull request? I would like to use the OData Connected Service as well in VS 2022.

**Comment by GoranHalvarsson** [comment]
Yes please. Please fix this pullrequest :-) So we can have ODataConnectedService in VS2022

**Review by 1Jesper1 — COMMENTED** [review]
MigrationBackup folder is obsolete?
Why is the packages.config file moved to MigrationBackup folder?

**Comment by 1Jesper1** [comment]
> @felix1000 I just tried to get your PR to work on my installation. I had to change this line to get the VSIX installer to install the new extension:
> 
> ```
> <Prerequisite Id="Microsoft.VisualStudio.Component.CoreEditor" Version="[15.0,18.0)" DisplayName="Visual Studio core editor" />
> ```
> 
> The installation succeeded, but I don't see the Odata option in the Connected Services view. However, Visual studio lists the extension as installed.
> 
> Any ideas?

I did not have that problem, I cloned the branch => Start debugging extension with F5 => New Visual studio 2022 opens => Go to extensions => Extension is visible in installed tab => Right click on project "Connected services" => Click on "Manage connected services" => Odata connected service is visable as options at "Other services". Seems good to me.

**Comment by henning-krause** [comment]
@1Jesper1 When I tested it using F5 (in debugging mode), everything worked as expected. I should have mentioned that in my initial post. However, when I installed  the extension using the vsix produced by Visual Studio, the addin was reported as being installed, but it did not show up under "Connected services".

**Comment by 1Jesper1** [comment]
> @1Jesper1 When I tested it using F5 (in debugging mode), everything worked as expected. I should have mentioned that in my initial post. However, when I installed the extension using the vsix produced by Visual Studio, the addin was reported as being installed, but it did not show up under "Connected services".

Thanks for the reply, I'm going to try tonight. How do I get the generated vsix?

**Comment by henning-krause** [comment]
@1Jesper1 Should be in the bin folder.

**Comment by 1Jesper1** [comment]
@henning-krause Thanks, I used your fix to install the extension!
It seems the Odata connected service is available after installing the extension, see: 
image

File: MigrationBackup/a6acfe1f/ODataConnectedService/NuGetUpgradeLog.html [hunk 1]
```diff
@@ -0,0 +1,193 @@
+<!DOCTYPE html>
```
Comment (habbes) [inline MigrationBackup/a6acfe1f/ODataConnectedService/NuGetUpgradeLog.html:1]: Should this file be included in source control?

File: src/ODataConnectedService/source.extension.vsixmanifest [hunk 2]
```diff
@@ -11,9 +11,15 @@
         <Tags>OData Connected Service</Tags>
     </Metadata>
     <Installation AllUsers="true">
-        <InstallationTarget Id="Microsoft.VisualStudio.Pro" Version="[15.0, 17.0)" />
-        <InstallationTarget Version="[15.0,17.0)" Id="Microsoft.VisualStudio.Community" />
-        <InstallationTarget Version="[15.0,17.0)" Id="Microsoft.VisualStudio.Enterprise" />
+        <InstallationTarget Id="Microsoft.VisualStudio.Pro" Version="[17.0,18.0)">
```
Comment (habbes) [inline src/ODataConnectedService/source.extension.vsixmanifest:14]: question: What was the reason for removing the older versions? We want to continue supporting VS 2017 and VS 2019. So ideally we want the targets `[15.0, 18.0)`

**Comment by habbes** [comment]
@felix1000 Thank you for contributing this PR.

I noticed the target versions to `[17.0, 18.0)`, this seems like it will drop support for VS 2017 and 2019 (15.x and 16.x respectively). We intend to continue supporting these versions for the foreseeable future. If it's impossible to build the projection with those targets, then ideally we should consider have two projects. One that builds for `[15.0, 17.0)` and one that builds for 17+.

**Comment by 1Jesper1** [comment]
@habbes I see the [PR](https://github.com/OData/ODataConnectedService/pull/247) has been merged?

**Comment by KenitoInc** [comment]
@1Jesper1 Since we merged the PR, kindly rebase this PR and fix conflicts

**Comment by 1Jesper1** [comment]
@KenitoInc I could, but I think I can't edit this PR to fix conflicts? Maybe @felix1000 ?

**Comment by pbvs** [comment]
,,

> Hello @pbvs I created #274 because I could not edit this PR. Could you test?

Hi @1Jesper1 thanks for your reply. I will try to test this right away and let you know. I will post my findings on #274. I will test the following points:
- Installation of vsix package
- Adding a Odata service regerence to a clean project
- Executing an odata request using the generated code

**Comment by 1Jesper1** [comment]
https://github.com/OData/ODataConnectedService/pull/295 merged, this is obsolete and can be closed.

**Comment by ElizabethOkerio** [comment]
@1Jesper @pbvs  @felix1000 @henning-krause you can try out this vsix for vs 2022 https://marketplace.visualstudio.com/items?itemName=marketplace.ODataConnectedService2022. Let us know if it works for you.

**Comment by 1Jesper1** [comment]
@ElizabethOkerio Link does not seem to work

**Comment by henning-krause** [comment]
@1Jesper1 https://github.com/OData/ODataConnectedService/releases is the correct link.

**Comment by henning-krause** [comment]
@ElizabethOkerio It works, however only in Visual Studio 2022 17.3 Preview, cause of this bug in Visual Studio: https://developercommunity.visualstudio.com/t/Connected-Services-plugins-cant-update-/10070285#T-ND10099077.

**Pull request closed without merging**